gcloud-java-dns/src/main/java/com/google/gcloud/dns/Zone.java

```diff
@@ -33,20 +36,90 @@
  * @see <a href="https://cloud.google.com/dns/zones/">Google Cloud DNS managed zone
  * documentation</a>
  */
-public class Zone implements Serializable {
+public class Zone extends ZoneInfo {
 
-  // TODO(mderka) Zone and zoneInfo to be merged. Opened issue #605.
+  private static final long serialVersionUID = -5817771337847861598L;
+  private final DnsOptions options;
+  private transient Dns dns;
 
-  private static final long serialVersionUID = 6847890192129375500L;
-  private final ZoneInfo zoneInfo;
-  private final Dns dns;
+  /**
+   * Builder for {@code Zone}.
+   */
+  public static class Builder extends ZoneInfo.Builder {
+    private final Dns dns;
+    private final ZoneInfo.BuilderImpl infoBuilder;
+
+    private Builder(Zone zone) {
+      this.dns = zone.dns;
+      this.infoBuilder = new ZoneInfo.BuilderImpl(zone);
+    }
+
+    @Override
+    public Builder name(String name) {
+      infoBuilder.name(name);
+      return this;
+    }
+
+    @Override
+    Builder id(String id) {
+      infoBuilder.id(id);
+      return this;
+    }
+
+    @Override
+    Builder creationTimeMillis(long creationTimeMillis) {
+      infoBuilder.creationTimeMillis(creationTimeMillis);
+      return this;
+    }
+
+    @Override
+    public Builder dnsName(String dnsName) {
+      infoBuilder.dnsName(dnsName);
+      return this;
+    }
+
+    @Override
+    public Builder description(String description) {
+      infoBuilder.description(description);
+      return this;
+    }
+
+    @Override
+    public Builder nameServerSet(String nameServerSet) {
+      infoBuilder.nameServerSet(nameServerSet);
+      return this;
+    }
+
+    @Override
+    Builder nameServers(List<String> nameServers) {
+      infoBuilder.nameServers(nameServers); // infoBuilder makes a copy
+      return this;
+    }
+
+    @Override
+    public Zone build() {
+      return new Zone(dns, infoBuilder);
+    }
+  }
+
+  Zone(Dns dns, ZoneInfo.BuilderImpl infoBuilder) {
+    super(infoBuilder);
+    this.dns = dns;
+    this.options = dns.options();
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
 
   /**
    * Constructs a {@code Zone} object that contains the given {@code zoneInfo}.
    */
   public Zone(Dns dns, ZoneInfo zoneInfo) {
-    this.zoneInfo = checkNotNull(zoneInfo);
-    this.dns = checkNotNull(dns);
+    super(new BuilderImpl(zoneInfo));
```

Comment:
Oh, you already have that toPb, so I think we should get rid of this method and update listZones as suggested